memcache/store.go

```diff
@@ -11,7 +13,11 @@ func (mcs *Store) Save() error {
 	item := acquireItem()
 	item.Key = provider.getMemCacheSessionKey(mcs.GetSessionID())
 	item.Value = value
-	item.Expiration = provider.expiration
+
+	if mcs.GetExpiration() > math.MaxInt32 {
```

Comment:
Override `Store.SetExpiration`in this provider to check this.

Reply:
Later, you could get expiration without checks.

Reply:
Can we safely consider this won't have any side effect regarding `store.HasExpirationChanged()`?
The expiration attribute is not exported so I'll need to call the parent `SetExpiration()` that can have side effects if called by someone else than the user.

Reply:
Yes `SetExpiration()` is call by user. But this check is not necessary all time, only when user change the expiration.

Therefore, overwrite `Store.SetExpiration` to memcache provider in store.go, like `Store.Save()`

Reply:
```go
func (mcs *Store) SetExpiration(expiration time.Duration) error {
    if expiration  > math.MaxInt32 {
        return errExpirationIsTooBig
    }
    mcs.newExpiration = expiration
    return nil
}
```

Reply:
In  Store type add again newExpiration field

Reply:
Ok, I see. Two questions though: 
1/ SetExpiration does not return an error in the interface. Should I add it?
2/ I think I also need to override the GetExpiration and HasExpirationChanged. Do you confirm?

Reply:
1) Yes
2) Why?

Reply:
1. OK
2. I'm a beginner in Go but if I'm not mistaken the attribute newExpiration in the new struct will not replace the original one from the parent struct, correct? If it's correct it means that GetExpiration and HasExpirationChanged should rely on the new attribute instead of the original from the parent struct. Where am I wrong? :)

Reply:
Ohh yes, you need to re-implement it in memcache store. Because if you don't do it, when you get expiration, returns original field not new field in memcache

Reply:
Ok, great, we are in sync.

Reply:
But it leads to other issues then... I need to add two attributes: `expiration` and `newExpiration` because I need to compare them in HasExpirationChanged. This leads to also override the Init method... which in turns leads to also override all other methods... What do you think?

Reply:
Oh no! Only re-add the expiration field like the newExpiration field, and overwrite methods in memcache store to get value from memcache store struct

Reply:
You don't need to override Init of provider.

Reply:
@savsgio , I saw that you put some comments in my mailbox but I cannot see them in the discussion. Did you remove them in the meantime?

Do you see my point? I'm talking about the `Init` method of the Store not the Provider. It's because if we add two attributes to the inherited store (expiration and newExpiration), we need to be able to initialize them using `Init` of the Store as it's done for the parent store of other providers. Otherwise, only the parent attributes will be correctly initialized. Do you see what I mean?

Reply:
Yes, i know.

Overwrite `Init` method too, because is the best solution i think.

Reply:
Done.